corporate/tests/test_stripe.py

```diff
@@ -5697,3 +5698,238 @@ def test_non_sponsorship_billing(self, *mocks: Mock) -> None:
+            "expected_total_users": "10 users",
+            "paid_users_count": "1 user",
+            "paid_users_description": "We have 1 paid user.",
+            "requested_plan": "Community",
+        }
+        response = self.client_billing_post("/billing/sponsorship", data)
+        self.assert_json_success(response)
+
+        customer = self.billing_session.get_customer()
+        assert customer is not None
+
+        sponsorship_request = ZulipSponsorshipRequest.objects.get(customer=customer)
+        self.assertEqual(sponsorship_request.requested_plan, data["requested_plan"])
+        self.assertEqual(sponsorship_request.org_website, data["website"])
+        self.assertEqual(sponsorship_request.org_description, data["description"])
+        self.assertEqual(
+            sponsorship_request.org_type,
+            Realm.ORG_TYPES["opensource"]["id"],
+        )
+
+        from django.core.mail import outbox
+
+        # First email is remote user email confirmation, second email is for sponsorship
+        message = outbox[1]
+        self.assert_length(outbox, 2)
+        self.assert_length(message.to, 1)
+        self.assertEqual(message.to[0], "sales@zulip.com")
+        self.assertEqual(message.subject, "Sponsorship request (Open-source project) for Zulip Dev")
+        self.assertEqual(message.reply_to, ["hamlet@zulip.com"])
+        self.assertEqual(self.email_envelope_from(message), settings.NOREPLY_EMAIL_ADDRESS)
+        self.assertIn("Zulip sponsorship request <noreply-", self.email_display_from(message))
+        self.assertIn(
+            "Support URL: http://zulip.testserver/activity/remote/support?q=demo.example.com",
+            message.body,
+        )
+        self.assertIn("Website: https://infinispan.org", message.body)
+        self.assertIn("Organization type: Open-source", message.body)
+        self.assertIn("Description:\nInfinispan is a distributed in-memory", message.body)
+
+        # Check /billing redirects you to sponsorship page.
+        response = self.client_get(f"{billing_base_url}/billing/", subdomain="selfhosting")
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response["Location"], f"/realm/{realm.uuid!s}/sponsorship/")
+
+        # Check sponsorship page shows sponsorship pending banner.
+        result = self.client_get(f"{billing_base_url}/sponsorship/", subdomain="selfhosting")
+        self.assert_in_success_response(
+            ["This organization has requested sponsorship for a", "Community"], result
+        )
+
+        # Approve sponsorship
+        billing_session = RemoteRealmBillingSession(
+            remote_realm=remote_realm, support_staff=self.example_user("iago")
+        )
+        billing_session.approve_sponsorship()
+        remote_realm.refresh_from_db()
+        self.assertEqual(remote_realm.plan_type, RemoteRealm.PLAN_TYPE_COMMUNITY)
```

Comment:
Later we can extend this to include having a support user login and approve the sponsorship via the API, but it's not critical.